SW3: add a way for Rules to target certain modes

henryaddison opened the pull request:
Some Rules that are being created need to be restricted to only being applied when validating data used in certain modes. For example, customer data on an Order is only present during one of the the C2 and B booking phases as opposed to the more general publishing of open data.

By default all Rules will target all modes to start with since mode targeting is optional and in many of the more generic rules won't need to be considered.

thtg88 commented on README.md:
```diff
@@ -228,6 +228,11 @@ const options = {
+
+Provides context as to usage mode that data should be validated against.
+For example, to only apply rules that are suitable for data used in a booking flow phase like C1, C2 or B.
+
```
@henryaddison 
Is it worth providing an example here like done in the other options' documentation?
Something along the lines of (if at all valid):
```js
const { validate } = require('@openactive/data-model-validator');

const model = {
  type: 'CustomAction'
  // ...
};

const options = {
  mode: ['C2', 'B']
};

const result = validate(model, options);

henryaddison replied:
Yes it would

henryaddison commented:
Note to self: use the updated list of modes C1Req, C1Response, C2Request, C2Response, PRequest, PResponse, BRequest, BResponse, OrderProposalPatch, OrderPatch, OrdersFeed, OrderStatus (from https://docs.google.com/spreadsheets/d/1Je-kraapkV8kpYEIjOspgsglZobefT4gBNNYNWZNxqM/edit#gid=2146889639). And is probably worth defining some sort of enum or constants to avoid typos.

henryaddison commented:
Have addressed those comments (mine & Marco's). My only concern is that `mode` is A) too similar to `model` so I made a lot of typos particularly with auto-complete and B) is a bit too generic.

thtg88 commented:
Very good observation - how about `validationMode`? Not very DRY but clearer than `mode`

nickevansuk commented:
`validationMode` sounds good

henryaddison commented:
Gone with validationMode & merging
